### PR TITLE
Ensure default (built in) commands are plugins.

### PR DESCRIPTION
**Pull request by ntoll**
All the sub-commands for `pyscript` should be implemented as pluggy plugins. To this end, this PR introduces:

* A `create` command that works.
* Tests for the _generator aspect of this command.
* Moved the `wrap` command to a plugin.
* Removed the `delete` (did nothing) and `version` (use `--version` instead) sub commands.
* Minor simplifications in a few places.

### CODE REVIEW

**Review by fpliger — APPROVED** [review]
Lgtm!  🚀

File: tests/test_generator.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Tests for utility functions in the _generator.py module that cannot be
+exercised because of limitations in the Typer testing framework (specifically,
+multiple "prompt" arguments).
+"""
+from unittest import mock
+
+from pyscript import _generator as gen
+
+
+def test_create_project() -> None:
```
Comment (mattkram) [inline tests/test_generator.py]: I refactored this to not need mocks. Please let me know what you think. I couldn't figure out how to easily do a suggestion with a new file (`conftest.py`)

**Review by mattkram — COMMENTED** [review]
LGTM. I just had a few comments/suggestions. Please let me know what you think :)

File: src/pyscript/_generator.py [hunk 2]
```diff
@@ -18,3 +21,29 @@ def file_to_html(input_path: Path, title: str, output_path: Optional[Path]) -> N
...
+) -> None:
+    """
+    New files created:
+
+    manifest.toml - project metadata
+    index.html - a "Hello world" start page for the project.
+
+    TODO: more files to add to the core project start state.
+    """
+    context = {k: v for k, v in locals().items() if not k.startswith("__")}
```
[the remark this answers is not in the document]
Comment (ntoll) [reply src/pyscript/_generator.py:40]: I was just being quick. Yes, more explicit is better. :-)

File: src/pyscript/_generator.py [hunk 3]
```diff
@@ -18,3 +21,38 @@ def file_to_html(input_path: Path, title: str, output_path: Optional[Path]) -> N
...
+    context = {
+        "app_name": app_name,
+        "app_description": app_description,
+        "author_name": author_name,
+        "author_email": author_email,
+        "version": f"{datetime.date.today().year}.1.1",
+        "created_on": datetime.datetime.now(),
+        "id": str(uuid4()),
+    }
+    context["version"] = f"{datetime.date.today().year}.1.1"
```
Comment (mattkram) [inline src/pyscript/_generator.py:49]: Just taking a peek and these three statements are already in the dict

**Comment by ntoll** [comment]
The changes you made are very welcome and definitely improve the PR.

File: src/pyscript/_generator.py (hunk 3 above)
Comment (ntoll) [reply src/pyscript/_generator.py:49]: Ack... lemme fix that.

**Pull request merged**